practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/service/PractitionerService.java

```diff
@@ -19,39 +21,65 @@
 @Service
 public class PractitionerService implements EntityService {
 
-    private EntityStore<Practitioner> practitionerStore;
-    private final ModelEntityValidation<Practitioner> entityValidation;
+    private final EntityStore<Practitioner> practitionerStore;
+    private final ModelEntityValidation<Practitioner> practitionerValidator;
+    @SuppressWarnings("FieldCanBeLocal")
     private final Logger logger = LoggerFactory.getLogger(PractitionerService.class);
 
+    private final EntityStore<RoleAssignment> roleAssignmentStore;
+    private final ModelEntityValidation<RoleAssignment> roleAssignmentValidator;
+
     /**
-     *
      * @param practitionerStore Practitioner store interface
-     * @param entityValidation Practitioner validation interface 
+     * @param practitionerValidator Practitioner validation interface
      */
     @Autowired
     public PractitionerService(EntityStore<Practitioner> practitionerStore,
-            ModelEntityValidation<Practitioner> entityValidation) {
+                               ModelEntityValidation<Practitioner> practitionerValidator,
+                               EntityStore<RoleAssignment> roleAssignmentStore,
+                               ModelEntityValidation<RoleAssignment> roleAssignmentValidator) {
         if (practitionerStore == null) {
             throw new InitialisationError("practitioner store cannot be null");
         }
-        if (entityValidation == null) {
-            throw new InitialisationError("entity validation cannot be null");
+        if (practitionerValidator == null) {
+            throw new InitialisationError("practitioner entity validation cannot be null");
+        }
+        if (roleAssignmentStore == null) {
+            throw new InitialisationError("RoleAssignment store cannot be null");
+        }
+        if (roleAssignmentValidator == null) {
+            throw new InitialisationError("RoleAssignment entity validation cannot be null");
         }
+
         this.practitionerStore = practitionerStore;
-        this.entityValidation = entityValidation;
+        this.practitionerValidator = practitionerValidator;
+
+        this.roleAssignmentStore = roleAssignmentStore;
+        this.roleAssignmentValidator = roleAssignmentValidator;
+
         logger.info(Services.STARTUP.message());
     }
 
     /**
-     *
      * @param practitioner the Practitioner to save.
      * @return The id of the new practitioner
      */
     public String savePractitioner(Practitioner practitioner) {
-        var validationResponse = entityValidation.validate(practitioner);
+        var validationResponse = practitionerValidator.validate(practitioner);
         if (!validationResponse.isValid()) {
             throw new ValidationException(validationResponse.getErrorMessage());
         }
         return practitionerStore.saveEntity(practitioner);
     }
+
+    //TODO: should this be on its own service?
```

Comment:
think so, what is in the role service right now?

Reply:
I think there's some misunderstanding here. I was referring to the "service" class within the practitioner-service. 
Currently the PractitionerService class also deals with RoleAssignments and I was think to separate that code into its own RoleAssignmentService class... I'm not sure about that, but anyway I would like to do it on its own task/PR to unblock a few other people that work on the practitioner-service module.